lib/transformers/set_default_initial_state.ex

```diff
@@ -0,0 +1,28 @@
+defmodule AshStateMachine.Transformers.SetDefaultInitialState do
+  # If there is only one value in inital_states we can set it
+  # as the default
+  @moduledoc false
+  use Spark.Dsl.Transformer
```

Comment:
we'll want to use 
```elixir
def before?(OtherTransformer), do: true
def before?(_), do: false
```

here ( or `def after` in the other one) to make sure they run in the proper order.

Reply:
I'll try by adding another case in AddState.